Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/format/xml/XMLReader.java

```diff
@@ -0,0 +1,573 @@
+            if (elementNS == null) {
+                elementNS = documentNS;
+            }
+            if (attributeNS.equals(elementNS)) {
+                result = element.getAttribute(attributeName);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Returns the type of the relation element or "" if this is not the case.
+     *
+     * @param element
+     *            elemnt to examine
+     * @param documentNS
+     *            the document namespace
+     * @return the type of the relation element, or ""
+     */
+    private static String getRelationElementType(Element element, String documentNS) {
+        if (!RDF.PREDICATE.getIdentifier().equals(getType(element, documentNS).orElse(null))) {
+            return "";
+        }
+        return getAttributeFromElement(element, RDF.VALUE.getIdentifier(), documentNS);
+    }
+
+    /**
+     * Returns the type URL of an attribute.
+     *
+     * @param attribute
+     *            attribute whose type is to return
+     * @param element
+     *            element the attribute is attached to
+     * @param documentNS
+     *            document namespace, as fallback
+     * @return the type URL
+     */
+    private static String getType(Attr attribute, Element element, String documentNS) {
+        String ns = attribute.getNamespaceURI();
+        if (ns == null) {
+            ns = element.getNamespaceURI();
+        }
+        String name = attribute.getLocalName();
+        if (ns == null) {
+            ns = documentNS;
+        }
+        return ns == null ? name : ns.endsWith("/") ? ns.concat(name) : ns + '#' + name;
+    }
+
+    /**
+     * Returns the type URL of an element.
+     *
+     * @param element
+     *            element whose type is to return
+     * @param documentNS
+     *            the address of the document, as fall-back namespace for local
+     *            XML elements and attributes
+     * @return the type URL
+     */
+    private static Optional<String> getType(Element element, String documentNS) {
+        String ns = element.getNamespaceURI();
+        if (ns == null) {
+            ns = documentNS;
+        }
+        String name = element.getLocalName();
+        if (ns == null) {
+            return Optional.of(name);
+        } else {
+            String type = ns.endsWith("/") ? ns.concat(name) : ns + '#' + name;
+            return Optional.ofNullable(type.equals(RDF.DESCRIPTION) ? null : type);
+        }
+    }
+
+    /**
+     * Returns the globally unique resource identifier (URI) for a local file as
+     * a {@code String}. The URI is formed with the host name of the machine the
+     * program is running on.
+     *
+     * @param file
+     *            file to create an URI for
+     * @return unique resource identifier for the file
+     * @throws IOException
+     *             if it fails
+     */
+    public static String globallyUniqueIdentifierForFile(File file) throws IOException {
```

Comment:
In Kitodo project files not accessed directly (except configuration files). For accessing files for example in the process directory of a process the module `Kitodo-FileManagement` should be used. Reason behind is that files are not necessary saved on a local file system and maybe provided be a media repository over http based requests. So maybe your self written method is not needed if you use the API from `Kitodo-FileManagement` module and URI objects.

Reply:
This is part of reading configuration XML files and rule sets, yes.

Reply:
As this is a public static method, who will prevent the usage for other use cases? I will hope that you are right and it is really only used for configuration files and not the meta data files itself.